src/hmem_ze.c

```diff
@@ -849,6 +854,10 @@ int ze_hmem_init(void)
 
 		dev_info[num_devices].uuid = dev_prop.uuid;
 
+		if (!dev_prop.subdeviceId)
+			pci_dev++;
+		dev_info[num_devices].device = pci_dev;
```

Comment:
This may not work as expected when ZE_AFFINITY_MASK is set.

Reply:
can find the correct pci device by comparing uuid.

Reply:
I'm not really sure how to find it with the uuid. That's a 16 byte uint8 array that seems to be unique for each tile. For example on my system:
```
{id = "\206\200\326\v/\000\000\000:\000\000\000\000\000\000\001"}
{id = "\206\200\326\v/\000\000\000:\000\000\000\000\000\000\002"}
{id = "\206\200\326\v/\000\000\000\232\000\000\000\000\000\000\001"}
{id = "\206\200\326\v/\000\000\000\232\000\000\000\000\000\000\002"}
```

Reply:
The bytes around the 9th byte are the pci address of the device. Not sure where is the start and end.
Note that ":" is 0x3a and "\232" is 0x9a. The pci addresses are `0000:3a:00.0` and `0000:9a:00.0`.

Reply:
Right I see that. But how do I map ":" and "\232" to 0 and 1? Is there a way to lookup all ZE PCI devices (regardless of the affinity mask)?

Reply:
@j-xiong See update which matches it based on the PCI bus ID. This isn't the prettiest thing ever but it does the job...